### PR TITLE
feat(backend): add user settings update feature

### PR DESCRIPTION
**Pull request by tuantran0910**
## Description

Add recent agent update feature.

## How Has This Been Tested?

I have defined those endpoints:

### `PATCH /users/settings`

- Check if the API is running.
- How to test ?

```bash
curl -X PATCH 127.0.0.1:5000/api/v1/users/settings -d '{"auto_scroll": false, "current_agent_id": "c262ab96-06a5-46eb-8120-3cf978d11119"}'
```

- Then, we get:

```bash
{
    "message": "Success",
    "headers": null,
    "data": {
        "current_agent_id": "c262ab96-06a5-46eb-8120-3cf978d11119",
        "auto_scroll": false
    }
}
```

### CODE REVIEW

File: chatbot-core/backend/app/services/user.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        """
+        if not recent_agent_ids:
+            recent_agent_ids = []
+        else:
+            recent_agent_ids = [
+                agent_id for agent_id in recent_agent_ids if agent_id != current_agent_id
+            ]
+
+        # Add the current agent ID to the beginning of the list
+        logger.info(f"Adding current agent ID: {current_agent_id} to the beginning of the list")
```
Comment (lelouvincx) [inline chatbot-core/backend/app/services/user.py:50]: Logging string interpolation uses f-string. Consider using structured logging.
```python
logger.info("Adding current agent ID to list", extra={"agent_id": current_agent_id})
```

**Review by lelouvincx — COMMENTED** [review]
The overall docstring and comments are vague and lack of information that makes me hard to understand. After all of this, please create another issue to add up docstrings to classes and methods.

**Comment by lelouvincx** [comment]
Is it ready to review again?

**Comment by tuantran0910** [comment]
> Is it ready to review again?

Yes

File: chatbot-core/backend/app/routers/v1/user.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+from app.utils.api.api_response import BackendAPIResponse
+from app.utils.api.error_handler import ErrorCodesMappingNumber
+from app.utils.api.helpers import get_logger
+from app.utils.user.authentication import get_current_user
+
+logger = get_logger(__name__)
+router = APIRouter(prefix="/users", tags=["user", "setting"])
+
+
+@router.patch("/settings", response_model=APIResponse, status_code=status.HTTP_200_OK)
```
[the remark this answers is not in the document]
Comment (lelouvincx) [reply chatbot-core/backend/app/routers/v1/user.py:22]: Ok we decide to use:
- `/users/me/settings` for all settings related to users
- That's all for one endpoint

**Review by lelouvincx — APPROVED** [review]
LGTM

**Pull request merged**